src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -715,10 +715,31 @@ public void restartContainer(String containerId, int secondsToWaitBeforeRestart)
 
   @Override
   public void killContainer(final String containerId) throws DockerException, InterruptedException {
+    checkNotNull(containerId, "containerId");
     containerAction(containerId, "kill");
   }
 
   @Override
+  public void killContainer(final String containerId, final TerminationSignal signal)
+      throws DockerException, InterruptedException {
+    checkNotNull(containerId, "containerId");
+
+    try {
+      final WebTarget resource = noTimeoutResource()
```

Comment:
why use the `noTimeout` client here?

seems like there is also room to re-use containerAction, by refactoring it to have an overload where you can pass additional query params, to maintain the idea that most of these methods that call `POST /containers/:id/:action` flow through the same method